DigitalLearningSolutions.Data/Services/DelegateDownloadFileService.cs

```diff
@@ -41,6 +68,30 @@ public byte[] GetDelegateDownloadFileForCentre(int centreId)
             return stream.ToArray();
         }
 
+        public byte[] GetAllDelegatesFileForCentre(
+            int centreId,
+            string? searchString,
+            string? sortBy,
+            string sortDirection,
+            string? filterString
+        )
+        {
+            using var workbook = new XLWorkbook();
+
+            PopulateAllDelegatesSheet(
+                workbook,
+                centreId,
+                searchString,
+                sortBy,
+                sortDirection,
+                filterString
+            );
+
+            using var stream = new MemoryStream();
+            workbook.SaveAs(stream);
+            return stream.ToArray();
+        }
+
         private void PopulateDelegatesSheet(IXLWorkbook workbook, int centreId)

```

Comment:
Do we need this method anymore? Does it not do the same thing as PopulateAllDelegatesSheet with no filters/search/sort?

Reply:
We do need it. It is not quite the same as PopulateAllDelegatesSheets. They use different columns. Additionally, PopulateDelegatesSheet populates the data different, as it uses the anonymous object approach so the column headers are just straight rips of the property names. The upload service does validation on those, and might depending on it in other ways if I recall, so it would be a bigger refactor, probably best suited to another ticket.